tests/nested/manual/cloud-init-never-used-not-vuln/task.yaml

```diff
@@ -0,0 +1,151 @@
+
+restore: |
+    #shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB/nested.sh"
+    destroy_nested_vm
+    cleanup_nested_env
+
+    apt remove -yqq genisoimage || true
+
+debug: |
+    if [ -f snapd-before-reboot.logs ]; then
+        echo "logs before reboot"
+        cat snapd-before-reboot.logs
+    fi
+    echo "logs from current nested VM boot snapd"
+    execute_remote "sudo journalctl -e --no-pager -u snapd" || true
+
+execute: |
+    #shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB/nested.sh"
+
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB/snaps.sh"
+
+
+    # the VM here will not ever had used cloud-init so snapd should disable 
+    # cloud-init when it is installed
+
+
+    # wait for done seeding
+    execute_remote "sudo snap wait system seed.loaded"
+
+    # if we are not building from current, then we need to prep the snapd snap
+    # to install with the fix, this simulates/verifies that devices in the field
+    # without the fix will actually be fixed after they refresh
+    if [ "$BUILD_FROM_CURRENT" = "false" ]; then
+        if is_core_16_nested_system; then
+            # build the core snap for this run
+            repack_snapd_deb_into_core_snap "$PWD"
+            copy_remote "$PWD/core-from-snapd-deb.snap"
+
+            # install the core snap
+            execute_remote "sudo snap install core-from-snapd-deb.snap --dangerous"
+
+            # now we wait for the reboot for the new core snap
+            wait_for_no_ssh
+            wait_for_ssh
+            
+        else
+            # build the snapd snap for this run
+            repack_snapd_deb_into_snapd_snap "$PWD"
+            copy_remote "$PWD/snapd-from-deb.snap"
+
+            # install the snapd snap
+            execute_remote "sudo snap install snapd-from-deb.snap --dangerous"
+        fi
+    fi
+
+    # Note: there is a race here after we have installed the fix (or we have 
+    # booted a fresh image with the fix). 
+    # Namely, snapd will begin checking on cloud-init status after it has 
+    # ensured that the boot was okay in the device manager, but this will not
+    # happen immediately in zero time, and moreover, snapd will not do anything
+    # substantial or measurable until cloud-init has reached a steady state or
+    # otherwise times out. 
+    # As such, in this test, we first wait for cloud-init to settle down, and 
+    # then wait a bit longer to give snapd a chance to run again and take 
+    # action that we can test for. Since in this test, cloud-init was never 
+    # used, the status command should not take much time at all, since it will 
+    # not have been triggered. 
+
+    echo "Waiting for cloud-init..."
+    execute_remote "cloud-init status --wait"
+
+    # TODO: is there a better thing we can wait for here instead? maybe the log
+    # message from snapd directly via journalctl ?
+    echo "Waiting for snapd to react to cloud-init"
+    sleep 60
```

Comment:
I think we could remove this step and directly check /etc/cloud/cloud-init.disabled or something similar, does it make sense?

Reply:
do you mean remove the sleep? we need something that waits a bit otherwise we will check too soon and snapd won't have restricted snapd yet, and so the test fails. We are effectively racing snapd to see if it restricts' cloud-init first before we check, or if we check before it restricts cloud-init. 

I have it on my TODO list to rewrite this sleep into a proper log check which will eliminate the race even on slow hardware. 

In any case if this is not critical, I would rather handle this in a followup so we can get master to where 2.45.2 is